src/main/java/Problem3/Problem3.java

```diff
@@ -7,5 +7,10 @@
 public class Problem3 {
     public static void printListInReverse(ListNode list, PrintStream print) {
         // homework
+        if (list != null && list.next != null) {
+            printListInReverse(list.next, print);
```

Comment:
Since you are using recursion, you only need to set one base case like:
`if (list == null) {
            return;
        }`